src/class/hid/hid_device.c

```diff
@@ -237,7 +237,7 @@ bool hidd_control_request(uint8_t rhport, tusb_control_request_t const * request
     }
     else if (request->bRequest == TUSB_REQ_GET_DESCRIPTOR && desc_type == HID_DESC_TYPE_REPORT)
     {
-      uint8_t const * desc_report = tud_hid_descriptor_report_cb();
+      uint8_t const * desc_report = tud_hid_descriptor_report_cb((uint8_t) request->wIndex);
```

Comment:
I think there is a bit of mixing-up here. The `wIndex` is actually the interface number of the HID interface and not the `desc_index`. It just happens to be the same 0, 1 order as the example. It will be different, if we have additional class before and/or between the hid interface.

What we should do here is use the wIndex to figure out the index of the targeted interface in hid pool and pass it as `itf` to the callback. It is a foor-loop comparing the wIndex agains the `itf_num` of the interface.
   
![image](https://user-images.githubusercontent.com/249515/94890737-4c8a1000-04aa-11eb-9a76-ac0b4f685409.png)

Reply:
We should be able to use the lower byte of wValue to determine the descriptor index. The setup packet has the correct index in it when sniffing with a beagle. I set up some traps to debug this and wValue is always 0x2200 instead of 0x2200 and then 0x2201. Trying to track down why, I feel like this is the right way to solve this problem if it isn't crazy. That way we are relying on the spec instead of a hack.

Reply:
![image](https://user-images.githubusercontent.com/3053809/94963781-443bd000-04be-11eb-86a7-c34bc4393755.png)
I wanted to include an image here. It is super wild, I have traced the data all the way back and can't figure out where it would be manipulated. It seems to just come in on the FIFO read so how would 1 byte be wrong on the setup packet.